src/partners/thorchain.ts

```diff
@@ -147,19 +148,43 @@ const makeThorchainPlugin = (info: ThorchainInfo): PartnerPlugin => {
         if (txStatus !== 'success') {
           continue
         }
-        if (pools.length !== 2) {
+
+        // Find the source asset
+        let srcAsset: string | undefined
+        let depositAmount: number | undefined
+        let srcAddress: string | undefined
+        for (const txIn of txIns) {
+          for (const coin of txIn.coins) {
+            if (pools.includes(coin.asset)) {
+              srcAsset = coin.asset
+              depositAmount = Number(coin.amount) / THORCHAIN_MULTIPLIER
+              srcAddress = txIn.address
+              break
+            } else {
+              if (pools.length === 1) {
+                // Either src or dest is the main asset (RUNE/CACOA)
+                srcAsset = coin.asset
+                depositAmount = Number(coin.amount) / THORCHAIN_MULTIPLIER
+                srcAddress = txIn.address
+              }
+            }
+          }
+          if (srcAsset != null) {
+            break
+          }
+        }
+        if (srcAsset == null || depositAmount == null) {
           continue
         }
 
-        const srcAsset = txIns[0].coins[0].asset
-        const match = txOuts.find(o => {
-          const match2 = o.coins.find(c => c.asset === srcAsset)
-          return match2 != null
+        const match = txOuts.some(o => {
+          const match2 = o.coins.some(c => c.asset === srcAsset)
+          return match2
         })
 
         // If there is a match between source and dest asset that means a refund was made
         // and the transaction failed
-        if (match != null) {
+        if (match) {
```

Comment:
Why did this condition invert, but not the comment?

Reply:
Condition didn't invert. The `match` var just become a boolean